libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestResponse.java

```diff
@@ -155,32 +154,12 @@ public void consumeQuietly() {
 		return responseAsBytes;
 	}
 
-    /**
-     * Checks if the response has a body.
-     *
-     * @return True - if response body is present, False - otherwise.
-     */
-	public boolean hasResponseBody() {
```

Comment:
That method was bogus - it's not checking for a body is checking for a response header content-type header set to json.